HousingRepairsOnlineApi/UseCases/RetrieveAvailableAppointmentsUseCase.cs

```diff
@@ -18,16 +20,32 @@ public RetrieveAvailableAppointmentsUseCase(IAppointmentsGateway appointmentsGat
             this.sorEngine = sorEngine;
         }
 
-        public async Task<IEnumerable<Appointment>> Execute(string repairLocation, string repairProblem, string repairIssue, string uprn)
+        public async Task<List<ApplicationTime>> Execute(string repairLocation, string repairProblem,
+            string repairIssue, string uprn)
         {
             Guard.Against.NullOrWhiteSpace(repairLocation, nameof(repairLocation));
             Guard.Against.NullOrWhiteSpace(repairProblem, nameof(repairProblem));
             Guard.Against.NullOrWhiteSpace(repairIssue, nameof(repairIssue));
             Guard.Against.NullOrWhiteSpace(uprn, nameof(uprn));
             var repairCode = sorEngine.MapSorCode(repairLocation, repairProblem, repairIssue);
+
+            var convertedResults = new List<ApplicationTime>();
+
             var result = await appointmentsGateway.GetAvailableAppointments(repairCode, uprn);
+            convertedResults.AddRange(result.Select(ConvertToHactAppointment));
+
+            return convertedResults;
+
+            ApplicationTime ConvertToHactAppointment(Appointment appointment)
+            {
 
-            return result;
+                return new ApplicationTime
+                {
+                    Id = appointment.Reference,
```

Comment:
In line with other comments about the type of `Id`, the value set here would need to be updated to reflect any changes made.